### PR TITLE
Fix: Automate blogpost imports with rss

### LINKED ISSUE #13462: Automatically import posts thru RSS feed
We currently post content to both https://paragraph.xyz/@unlockprotocol and https://unlock-protocol.com/blog
We want to avoid having to do that step twice. For this, we should have a mechanism that automatically import posts from Paragraph and adds them automatically to our git repo inside https://github.com/unlock-protocol/unlock/tree/master/unlock-protocol-com/blog

The steps to do that:
* create a Github action that runs once per day
* This action will fetch the RSS feed from Paragraph https://paragraph.xyz/api/blogs/rss/@unlockprotocol
* And for each post, it will create a [new file this folder](https://github.com/unlock-protocol/unlock/tree/master/unlock-protocol-com/blog) with the right _header_ section (`title`, `subtitle`, `authorName`, `publishDate`, `description` and `image`.
* the script should also fetch any image in the posts and store them locally in a [subfolder of this one](https://github.com/unlock-protocol/unlock/tree/master/unlock-protocol-com/public/images/blog) and replace the image URL to point to the local images (no hotlinking!)
* Once done, the script should open a PR automatically to make sure these posts are added to the git repo!

### PR DESCRIPTION
**Pull request by njokuScript**
# Description
This PR fixes the blogposts automation

- [x] Include mock for testing script
- [x] Check for duplicate blogs 

# Issues

Fixes #13462 13462
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [x] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — COMMENTED** [review]
Can you please remove the python script?

File: .github/actions/blog/action.yml [hunk 1]
```diff
@@ -13,18 +13,13 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Set up Python
-        uses: actions/setup-python@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
         with:
-          python-version: '3.x'
+          node-version: '16'
```
Comment (julien51) [inline .github/actions/blog/action.yml:19]: Ca you please use node 20 like the rest of the project please?

File: .github/actions/blog/rss_feed.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        slug
+      )
+      fs.ensureDirSync(postImagesDir)
+
+      // Fetch and save the image locally
+      let localImagePath = ''
+      if (imageUrl) {
+        const imageFilename = path.basename(imageUrl)
+        localImagePath = path.join(postImagesDir, imageFilename)
+        fetch(imageUrl)
```
Comment (julien51) [inline .github/actions/blog/rss_feed.js:64]: I realize you are *not* extracting the images from the content. Please add this to make sure we are not hotlinking images.

**Comment by njokuScript** [comment]
> Can you please remove the python script?

attended to the comments

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (julien51) [inline .eslintrc.json:1]: Please don't add a new eslint config. We already have a package fort that!

File: .github/actions/blog/action.yml [hunk 4]
```diff
@@ -13,18 +13,13 @@ jobs:
...
 
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install feedparser requests
-
-      - name: Fetch RSS feed and create posts
-        run: python rss_feed.py
+      - name: Run RSS Feed Script
+        run: node rss_feed.js
```
Comment (julien51) [inline .github/actions/blog/action.yml:22]: If you don't install the deps this won't run

**Review by julien51 — COMMENTED** [review]
Can you add the trigger `workflow_dispatch` so we can test this? I have a suspicion this would not run :(

**Comment by julien51** [comment]
ok so I forked your branch and run it locally and ... there are issues:
Screenshot 2024-03-18 at 5 22 14 PM

**Comment by njokuScript** [comment]
> ok so I forked your branch and run it locally and ... there are issues: Screenshot 2024-03-18 at 5 22 14 PM

the issue here is with the image not displaying properly right?

**Comment by julien51** [comment]
> the issue here is with the image not displaying properly right?

Not just the images, the author name is missing too. And looking at the posts, the images are also missing in there.

**Comment by julien51** [comment]
Please test this _extensively_ to make sure it works as expected.

File: package.json [hunk 5]
```diff
@@ -12,11 +12,13 @@
     "@typescript-eslint/eslint-plugin": "7.1.0",
     "@typescript-eslint/parser": "7.1.0",
     "babel-plugin-polyfill-corejs2": "0.4.8",
+    "feedparser-promised": "2.0.1",
     "typescript": "5.3.3"
   },
   "devDependencies": {
     "eslint": "8.54.0",
     "eslint-plugin-evm-address-to-checksummed": "0.0.6",
+    "eslint-plugin-import": "^2.25.2",
```
Comment (julien51) [inline package.json:21]:
```suggestion
    "eslint-plugin-import": "2.25.2",
```

**Comment by julien51** [comment]
@njokuScript Any news on that?

**Comment by njokuScript** [comment]
> @njokuScript Any news on that?

sorry, was travelling. Just getting to it now! will update in 2 hours.

**Comment by julien51** [comment]
ha! Thanks for the heads up! Ok I will look into it!

**Comment by njokuScript** [comment]
> > @julien51 I'm currently occupied with a tool I'm working on: https://github.com/njokuScript/base-migrate
> > Can someone else pick this up from where I left? This is to ensure that I'm not a blocker on this.
> 
> This is quite interesting! We have been looking at #11676 so maybe we can use your tool! wdyt @clemsos ?

That's perfectly what the tool does. It automates the process. We're launching on Base on Friday and we'll support more chains in coming days as we get feedback/feature requests

**Pull request closed without merging**